rust: Mark all public struct/enum as `non_exhaustive`

cathay4t opened the pull request:
All the public struct/enum in nmstate might add members/type, hence
marking them as `non_exhaustive` which will prevent API user from
in-compatibility usage.

Please refer to https://doc.rust-lang.org/reference/attributes/type_system.html
for more detail.

ffmancera reviewed (changes requested):
If we are going to mark all public structs and enums as non-exhaustive, then this patch is missing some of them like:

`NetworkState`, `DnsState`, `DnsClientState`, `NmstateError`, `InterfaceType`, `Interface`, `InterfaceIpv4`, `InterfaceIPv6`.. and more. Please, check them if possible. Thank you!

cathay4t commented:
Fixed.

ffmancera reviewed (changes requested):
For nmstate-rust code it looks good. But I think we should do the same for `libnm_dbus` public structs and enums. What do you think?

cathay4t committed:
rust: Mark all public struct/enum as `non_exhaustive`

All the public struct/enum in nmstate might add members/type, hence
marking them as `non_exhaustive` which will prevent API user from
in-compatibility usage.

Please refer to https://doc.rust-lang.org/reference/attributes/type_system.html
for more detail.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
libnm_dbus fixed.

ffmancera reviewed (approved):
LGTM!
